compiler/rustc_codegen_ssa/src/base.rs

```diff
@@ -486,6 +487,30 @@ fn get_argc_argv<'a, 'tcx, Bx: BuilderMethods<'a, 'tcx>>(
     }
 }
 
+pub fn get_debugger_visualizers(
+    tcx: TyCtxt<'_>,
+    visualizer_type: DebuggerVisualizerType,
+) -> FxHashSet<DebuggerVisualizerFile> {
+    if tcx.sess.crate_types().iter().all(|crate_type| crate_type.is_archive()) {
```

Comment:
@michaelwoerister 

No, I don't believe we do since the visualizers will still live in the crate metadata for the static libs. In the case of MSVC targets, the visualizers can only be embedded in the PDB which does not get created for a static lib. I'm not as familiar with linux targets but does this work differently? Is it sufficient to only embed the transitive visualizers in a leaf crate such as an executable, since they will be linking against all of the static libs anyway.

Reply:
As per my previous comment, I'll reach out to offline to resolve this issue as well.

Reply:
The name `CrateType::StaticLib` might be a bit misleading. Static libs are not Rust libraries anymore and contain no crate metadata. One uses them if one needs to produce a static library that looks like a C library from the outside (similar to `cdylib`). See https://doc.rust-lang.org/reference/linkage.html.

So as far as the Rust compiler is concerned, static libs *are* leaf crates. In the case of PDBs, there's not much we can do, I think. The build system that will eventually invoke the linker with the static lib as input needs to take care of adding the relevant NatVis files (unless there's some feature that allows to bundle NatVis files in regular `.lib` files).

But for ELF binaries we can just add the pretty printers as a normal section of the object files.